<launch>

  <!-- Basic data -->
  <arg name="robot_namespace"                    default="drone1" />
  <arg name="catalog_path"                       default="$(find behavior_coordinator)/src/test/behavior_catalog.yaml" />
  <arg name="testing"                            default="true" />

  <!-- topics & services-->
  <arg name="start_task_srv"                     default="start_task" />
  <arg name="stop_task_srv"                      default="stop_task" />
  <arg name="behavior_activation_finished_sub"   default="behavior_activation_finished" />
  <arg name="task_stopped_pub"                   default="task_stopped" />
  <arg name="list_of_running_tasks_pub"          default="list_of_running_tasks" />

  <group ns="$(arg robot_namespace)">

    <!-- Basic data -->
    <param name="robot_namespace"     value="$(arg robot_namespace)"    type="str" />
    <param name="catalog_path"        value="$(find behavior_coordinator)/src/test/behavior_catalog.yaml" />
    <param name="testing"             value="true" />

    <!-- topics & services -->
    <param name="~start_task_srv"                         value="$(arg start_task_srv)"                     type="str" />
    <param name="~stop_task_srv"                          value="$(arg stop_task_srv)"                      type="str" />
    <param name="~behavior_activation_finished_sub"       value="$(arg behavior_activation_finished_sub)"   type="str" />
    <param name="~task_stopped_pub"                       value="$(arg task_stopped_pub)"                   type="str" />
    <param name="~list_of_running_tasks_pub"              value="$(arg list_of_running_tasks_pub)"          type="str" />
    
    <node name="behavior_coordinator" pkg="behavior_coordinator"  type="behavior_coordinator"  output="screen"/>
    <test test-name="behavior_coordinator_test" pkg="behavior_coordinator" type="behavior_coordinator_test" />
  </group>

</launch>
